src/dll/Hooks/ValidateScripts.cpp

```diff
@@ -89,3 +89,59 @@ bool Hooks::ValidateScripts::Detach()
     isAttached = result != NO_ERROR;
     return !isAttached;
 }
+
+std::wstring WritePopupMessage(const std::vector<ValidationError>& validationErrors,
+                               const std::vector<PluginSystem::PluginName>& incompatiblePlugins)
+{
+    if (validationErrors.empty())
+    {
+        return {};
+    }
+
+    std::wstring message;
+
+    if (!incompatiblePlugins.empty())
+    {
+        fmt::format_to(std::back_inserter(message),
+                       L"The following red4ext plugins could not be loaded because they are "
+                       L"incompatible with the current version of the game:\n");
+
+        for (const auto& plugin : incompatiblePlugins)
+        {
+            fmt::format_to(std::back_inserter(message), L"- {}\n", plugin);
+        }
+        fmt::format_to(std::back_inserter(message), L"\n");
+    }
+
+    std::unordered_set<std::string_view> faultyScriptFiles;
+
+    for (const auto& error : validationErrors)
+    {
+        const auto ref = error.GetSourceRef();
+        if (ref)
+        {
+            faultyScriptFiles.insert(ref->file);
```

Comment:
Is this string_view still valid after the scope?

Reply:
@WopsS it's managed by `SourceRefRepository`, so it lives as long as the repository (forever)

Reply:
Ah, true. For some reason I read it as it is returned from the redscript's API.